condor_watch_q.py

```diff
@@ -691,15 +726,25 @@ def find_job_event_logs(
                 )
                 already_warned_missing_log.add(cluster_id)
             continue
-
         # if the path is not absolute, try to make it absolute using the
         # job's initial working directory
-        if not os.path.isabs(log_path):
-            log_path = os.path.abspath(os.path.join(ad["Iwd"], log_path))
+
+        if "DAGManNodesLog" in ad:
+            dagman_clusters_to_path[cluster_id] = ad["DAGManNodesLog"]
+            log_path = ad["DAGManNodesLog"]
+        else:
+            if not os.path.isabs(log_path):
+                log_path = os.path.abspath(os.path.join(ad["Iwd"], log_path))
```

Comment:
Let's collapse this into an `elif`... but also this is making the logic look kind of weird, because we unconditionally set `log_path` above. I suggest doing as pass on this block and the block above where `log_path` gets set and trying to combine them all to make the logic less convoluted.

Reply:
Logic in this function still needs a rework, I think.